kagami/webui/src/api/resourceDetail.ts

```diff
@@ -1,49 +1,41 @@
 import api, { ApiError } from './api';
 
-export enum Resourcestatus{
-    ONLINE = "ONLINE",
-    OFFLINE = "OFFLINE",
-    ERROR = "ERROR"
-
-};
 
+// 定义资源详情类型
 export type ResourceDetail = {
-    name:string;
-    upstream_url:string;
-    workers:{
-        name:string;
-        replica_id:number;
-        provider_method:string;
-        status:Resourcestatus
-    }
+    name: string; // 资源名称
+    providers: { // 提供者信息数组
+        replica_id: number; // 副本 ID
+        upstream_url: string; // 上游 URL
+        provider_method: string; // 提供方法
+        status: string; // 提供者状态
+    }[]; // 数组类型
 };
 
+// 获取资源详情的函数
 export const getResourceDetail = async (resource_name: string): Promise<ResourceDetail> => {
-    const response = await api.get(`/resource_detail/${resource_name}`);
+    const response = await api.get(`/resource/${resource_name}`);
     const data = response.data;
 
-    // Validate the response data structure
+    // 验证数据是否为对象类型
     if (!data || typeof data !== 'object') {
         throw new ApiError(400, 'INVALID_RESPONSE', 'Invalid resource detail data format', data);
     }
 
-    // Validate required fields
-    if (!data.name || !data.upstream_url || !data.workers) {
+    // 验证所需字段是否存在
+    if (!data.name || !data.providers || !Array.isArray(data.providers)) {
         throw new ApiError(400, 'MISSING_FIELDS', 'Resource detail is missing required fields', data);
     }
 
-    // Validate workers data
-    if (!Array.isArray(data.workers)) {
-        throw new ApiError(400, 'INVALID_WORKERS', 'Workers data must be an array', data);
-    }
-
-    // Validate each worker's data structure
-    for (const worker of data.workers) {
-        if (!worker.name || 
-            typeof worker.replica_id !== 'number' || 
-            !worker.provider_method || 
-            !Object.values(Resourcestatus).includes(worker.status)) {
-            throw new ApiError(400, 'INVALID_WORKER_DATA', 'Worker data is invalid or missing required fields', worker);
+    // 验证每个 provider 的结构
+    for (const provider of data.providers) {
+        if (
+            typeof provider.replica_id !== 'number' ||
+            typeof provider.upstream_url !== 'string' ||
+            typeof provider.provider_method !== 'string' ||
+            typeof provider.status !== 'string'
+        ) {
+            throw new ApiError(400, 'INVALID_PROVIDER_DATA', 'Provider data is invalid or missing required fields', provider);
```

Comment:
You can read [this](https://juejin.cn/post/7166061734803963917) for a better way to validate reponses.

Reply:
The API documentation should specify when each field exists and what data type it should be. If both the frontend and backend strictly follow the documentation, there is no need for excessive verification.

cc @wxiwnd , please review.

Reply:
This api definition had some changes. Please update the code  to apifox documentation.

Reply:
btw, please rebase the code.